rama-tcp/src/client/connect.rs

```diff
@@ -23,6 +23,49 @@ use tokio::{
     },
 };
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum IPModes {
+    Dual,
+    SingleIpV4,
+    SingleIpV6,
+    DualPreferIpV4
+}
+
+impl Default for IPModes {
+    fn default() -> Self {
+        Self::Dual
+    }
+}
+
+//DNS Resolver
```

Comment:
just a note for future commits: do put some effort in these comments once you are preparing for final review please, if you need help do let me know. Not important at this stage though, just remarking it already.

Reply:
I was in two minds whether to commit as  I was still thinkig about the IPMode matching implementation. I will keep note of that. Thanks